src/test/scala/mesosphere/marathon/api/v2/AppsResourceTest.scala

```diff
@@ -1023,13 +1022,38 @@ class AppsResourceTest extends MarathonSpec with MarathonActorSupport with Match
         |}""".stripMargin.getBytes("UTF-8")
 
     When("The application is updated")
-    val response = appsResource.replace(app.id.toString, body, force = false, auth.request)
+    val response = appsResource.replace(app.id.toString, body, force = false, partialUpdate = true, auth.request)
 
     Then("The return code indicates success")
     response.getStatus should be(200)
     response.getMetadata.containsKey(RestResource.DeploymentHeader) should be(true)
   }
 
+  test("Replacing an existing docker application, upgrading from host to user networking") {
+    Given("A docker app using host networking and non-empty port definitions")
+    val app = AppDefinition(
+      id = PathId("/app"), container = Some(Container.Docker(image = "foo")), portDefinitions = PortDefinitions(0))
+
+    When("upgraded to user networking using full-replacement semantics (no port definitions)")
+    val body =
+      """{
+        |  "cmd": "sleep 1",
+        |  "container": {
+        |    "type": "DOCKER",
+        |    "docker": {
+        |      "image": "/test:latest",
+        |      "network": "USER"
+        |    }
+        |  },
+        |  "ipAddress": { "name": "dcos" }
+        |}""".stripMargin.getBytes("UTF-8")
+    val appUpdate = appsResource.canonicalAppUpdateFromJson(app.id, body, false)
```

Comment:
should we also include a test case where `partialUpdate=true` to demonstrate that it will fail validation?